src/main/java/com/flightstats/hub/spoke/FileSpokeStore.java

```diff
@@ -178,14 +181,19 @@ private void keysInBucket(String key, OutputStream output) {
         }
         try {
             Collection<File> files;
-            if (resolution.equals("second")) {
+            if (Optional.ofNullable(resolution).orElse("FALSE").equals("second")) {
                 // filter all files in the minute folder that start with seconds
                 FileFilter fileFilter = new WildcardFileFilter(SpokePathUtil.second(key) + "*");
-                files = Arrays.asList(directory.listFiles(fileFilter));
+
+                files = Arrays.asList(Optional.ofNullable(directory.listFiles(fileFilter)).orElse(new File[]{}));
             } else {
                 files = FileUtils.listFiles(new File(path), null, true);
             }
-            for (File aFile : files) {
+            Collection<File> permanentFiles = files
+                    .stream()
+                    .filter(this::doesNotEndWith)
+                    .collect(Collectors.toList());
+            for (File aFile : permanentFiles) {
```

Comment:
Unrelated to this line, but related to this method...

The name of this method misled me to think it should have some sort of return value (like the other `keysInBucket` method in this class). Would you be willing to rename this method to `readKeysInBucket`? The calling method with that name seems to just call this method without any sort of additional behavior whatsoever. You'll need to change the access level on this, since I think the other method's not private, though.

Reply:
done